### DISCUSSION TITLE
Clean work dir without sudo

### DISCUSSION THREAD

**Discussion by akotlar** [nextflow-io/nextflow#5487] [Q&A]
nextflow clean <run-name> -f , or indeed manual `rm -rf /path/to/work/dir` fails because it appears that any of the files that were generated within a docker container are "root". Using "sudo" is a nonstarter, since that would prevent the command from running headless.

Presumably, if I could issue the rm -rf command from within a nextflow process that is executed in Docker after all outputs are copied to the --output directory, I could get around this issue, but I cannot figure out how to do this. Any help would be appreciated.

**Comment by colindaven** [comment]
I would use a singularity or apptainer container built from your Dockerfile or docker container instead, that way you can avoid this problem. Running anything which can generate files as root on a critical shared machine is not good practice - which is why many hpcs do not allow docker.

**Comment by pditommaso** [comment]
Also, I believe, podman does much better to address this problem

**Reply by akotlar** [reply]
Cool, thanks guys
